Python/wrfrun/fixIM.py

```diff
@@ -0,0 +1,172 @@
+# ================================ Pre-requisites =================================
+# =================================================================================
+
+pfx_f = "FILE-F:"                                                              # Pywinter output IM file prefix.
+fileout = "FILEOUT"                                                            # Output file name ("FILEOUT"'s a convention in pyWPS).
+SEAICE_fill_missing = -1E30                                                    # Value to use to fill SEAICE missing points.
+[stlat_a, stlon_a] = [86.95000457763672,-199.20001220703125]                   # Correct (grib1) IM start lat and start lon. 
+sl_layers = ['000007','007028','028100','100289']                              # Soil layers (cm-cm).
+plevs = np.array([                                                             # Grib pressure levels (Pa).
+  100000.0, 97500.0, 95000.0, 92500.0, 90000.0, 87500.0, 85000.0, 82500.0,               
+  80000.0,  77500.0, 75000.0, 70000.0, 65000.0, 60000.0, 55000.0, 50000.0, 
+  45000.0,  40000.0, 35000.0, 30000.0, 25000.0, 22500.0, 20000.0, 17500.0, 
+  15000.0,  12500.0, 10000.0, 7000.0,  5000.0,  3000.0,  2000.0,  1000.0, 
+  700.0,    500.0,   300.0,   200.0,   100.0])          
+
+
+# ===============================================================================
+# ================================= Main program ================================
+# ===============================================================================
+    
+if __name__ == '__main__':
+
+  # Retreive date and pl and sl grib files' and IM file's names (and paths)  
+  IMDate = argv[1]
+  Grb_pl = argv[2]
+  Grb_sl = argv[3]
+  IMF = argv[4]
+  
+  # Prompt on screen
+  print('\n================================== Processing ' + IMDate + ' ==================================')
+  
+  # Open IM file
+  interf = pyw.rinter(IMF)
+    
+  # Display keys
+  print('\nInput IM keys =',interf.keys())  
+  
+  # Get IM SEAICE values
+  IM_SEAICE = interf['SEAICE'].val
+  
+  # Open grib SL file & read seaice values
+  grbs = pygrib.open(Grb_sl)
+  grb = grbs.select(name='Sea ice area fraction')[0]
+  GRB_SEAICE = grb.values
+  
+  # Check that IM_SEAICE and GRB_SEAICE arrays are the same shape
+  if (IM_SEAICE.shape!=GRB_SEAICE.shape):
+    raise ValueError("IM_SEAICE.shape!=GRB_SEAICE.shape. "
+      "Intermediate and grib SEAICE arrays should be the same size.") 
+  
+  # Retreive stalt and stlon
+  stlat = interf['UU'].geoinfo['STARTLAT']
+  stlon = interf['UU'].geoinfo['STARTLON']
+  dlat = interf['UU'].geoinfo['DELTALAT']
+  dlon = interf['UU'].geoinfo['DELTALON']   
+  # NOTE: Here we assume all fields have the same stlat, etc.
+  
+  # Fix stlat and stlon values, if needed 
+  if ((stlat!=stlat_a) or (stlon!=stlon_a)):
+    stlat = stlat_a
+    stlon = stlon_a
+    print('\nstlat,stlon != stlat_a.stlon_a, so fixed the values.') 
```

Comment:
print the original values and what was substituted; if the original values are predictable (predictably wrong), you could do 
```
if stlat == stlat_expected and stlon == stlon_expected:
    stlat = stlat_a
    stlon = stlon_a
    print('informative message with substituted values')
else:
    raise NotImplementedError('print informative message here')
```
If the values are not always binary-identical, you can use np.allclose; for a quick-and-dirty implementation, np.testing.assert_almost_equal is also an option.

Reply:
I think that the original values are not predictable. Nonetheless, I added the printing of the original and the substituted values. Also, considering that all the grib1 and grib2 array values are binary equal and IF THIS OPTION IS ON, we want the stlat and stlon values to be equal in double precision, I think that binary check would be Ok here. However, I forgot to consider the possibility of 360 differences in lons. I will add that.

Reply:
Yes, binary check is correct here.
Are you doing another commit for the 360 issue? probably not that important, though...

Reply:
Alright. I did fix the 360 issue already (see the second commit). In particular it has:

"if ((stlat!=stlat_a) or (((np.abs(stlon-stlon_a))%360.0)!=0.0)):"